### DISCUSSION TITLE
Solver not converging -serpentinisation (2)

### DISCUSSION THREAD

**Discussion by AndyB-NHSG** [reaktoro/reaktoro#404] [General]
I have just started using reaktoro (I'm an old EQ3/6 user) and  been looking at essentially the same system as @erexshah, using a slightly modified version of his code, and see similar issues. I have created a local version of the supcrt database to which I have added ΔGf data from  the McCollom & Bach paper for minnesotaite and greenalite as well as creating a data block for Fe(OH)2, and I also added to the code to enable brucite as a solid solution.
I can get the code to converge using various combination of the solid solutions olivine, orthopyroxene, serpentine, talc and brucite  but four solid solutions seems to be some kind of tipping point. 

- If I run the code with olivine, serpentine and talc converges fine giving me talc at higher Ts but not as temperature is reduced - as expected. 
- Adding brucite  solid solution (to the olivine, talc & serpentine) I get convergence at high temperatures (when brucite isn't stable and so remains at 1e-16mol). As I run at lower temperatures, when brucite comes into he assemblage the solver doesn't converge. 
- Taking talc out of the system (i.e. olivine, serpentine, brucite) at the same temperature results in a stable solution without all the water being consumed. 
- It isn't just systems with 2 hydrous minerals -running with olivine, orthopyroxene, serpentine and brucite at 300C again results in the water being consumed and no solution.
Any thoughts?

Andy

**Comment by allanleal** [comment]
Hi @AndyB-NHSG - thanks for sharing the information above and a summary of your findings so far. I'm currently working with @erexshah on this problem. If not exactly the same problem, I'd suggest that you share here, or via email, the Python code you produced for these calculations (with custom database) so that I can have a look.

**Reply by AndyB-NHSG** [reply]
Thanks Allan. I think it is the same problem as described by @erexshah. I've used code very similar to his; I'll email you the custom database file for reference.

Andy
